numpy/core/src/umath/override.c

```diff
@@ -563,18 +562,24 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
                     Py_DECREF(errmsg);
                 }
             }
+            Py_DECREF(override_args);
             goto fail;
         }
 
         /*
          * Set the self argument of our unbound method.
          * This also steals the reference, so no need to DECREF after.
          */
-        PyTuple_SetItem(override_args, 0, override_obj);
+        if (PyTuple_SetItem(override_args, 0, override_obj) < 0) {
```

Comment:
Now that we're moving the creation of the override tuple, I would suggest to create the whole tuple here, and thus remove the (unpleasant) setting of the first element to `Py_None`. It also means it will not get unnecessarily constructed (e.g., if a class and a subclass both define `__array_ufunc__`).